SU2_CFD/src/drivers/CDiscAdjMultizoneDriver.cpp

```diff
@@ -286,36 +295,31 @@ void CDiscAdjMultizoneDriver::Run() {
        *    (Solution might be overwritten when entering another zone because of cross derivatives.) ---*/
 
       Set_BGSSolution(iZone);
-        
-      /*--- make sure that everything is loaded into the output container ---*/
-      
-      output_container[iZone]->SetHistory_Output(geometry_container[iZone][INST_0][MESH_0],solver_container[iZone][INST_0][MESH_0], config_container[iZone]);
-      
-    }
 
-    /*--- Now all coupling terms are summed up, set External_Old to External for next outer iteration. ---*/
+      /*--- Now all iZone coupling terms are summed up, set External_Old to External so the
+       *    next zone receives the contributions from iZone (BGS-type iteration). ---*/
+
+      Set_OldExternal();
```

Comment:
@oleburghardt the residuals for the cht case changed probably due to moving this call into the loop over zones, which as the effect of carrying the cross-dependencies to the next zone earlier. If you could make sure all is ok it would be great.

Reply:
> External still accumulates over all zones
Yes but it has to accumulate over all zones for all zones.

nZone=2 in fact is not the decisive thing but one would have to require that a set of solution variables has influence on at most one other zone. 
Which would be alright to assume for the types of couplings that we have but it's not as general (and hard to debug once we are there).

Imagine a case with 3 zones and each one has an influence on all the others. Then at iZone=1, we would update adjoints in zone 1 *and* overwrite External for zone 2 and 3 with cross dependencies given through solvers in zone 1.
 But External in zone 2 could also have had an contribution from a possible dependence of solvers in zone 3 on the solution in zone 2.

Reply:
> Setting external_old = external does not reset the contents of external.

Yes we accumulate in `External` (for each combination of `iZone` and `jZone`) but cross contributions only enter the solution through `External_Old`.
So by the above example, if they are overwritten in iZone, they might be incorrect in iZone+1 if there is a third party.

> Unless I'm missing something, do we have an all-to-all multizone case to put this to the test?

Unfortunately not, we have a 4-zone case but only with at most one off-diagonal block per zone.